kartothek/io_components/metapartition.py

```diff
@@ -1600,8 +1602,42 @@ def parse_input_to_metapartition(obj, metadata_version=None):
             metadata_version=metadata_version,
         )
     elif isinstance(obj, MetaPartition):
-        return obj
+        mp = obj
     else:
         raise ValueError("Unexpected type: {}".format(type(obj)))
 
+    # Check whether index columns of `mp` are equal to `assert_secondary_indices` to prevent
+    # metadata errors
+    if assert_partition_level_indices is not False:
+        if len(mp.metapartitions) > 1:  # nested metapartition
+            for _mp in mp.metapartitions:
```

Comment:
1. Please don't loop over the metapartitions attribute explicitly but rather loop over the `MetaPartition` instance itself. There is a MetaPartitionIterator which takes care of this for you
2. Instead of looping here you should verify the index in the above code branch in `isinstance(obj, dict)`. Otherwise we'll verify this in every recursion call and for input which doesn't even support indices (e.g. DF only)

Reply:
Rookie mistake, my bad